core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -74,13 +74,12 @@ class SparkHadoopUtil extends Logging {
     }
   }
 
-  /**
-   * Return an appropriate (subclass) of Configuration. Creating config can initializes some Hadoop
-   * subsystems.
-   */
-  def newConfiguration(conf: SparkConf): Configuration = {
-    val hadoopConf = new Configuration()
 
+  /**
+    * Appends S3-specific configurations and spark.hadoop.* configurations to a Hadoop
```

Comment:
Maybe also mention the buffer keys that are also added that don't fit into either of those.